skimage/metrics/_structural_similarity.py

```diff
@@ -252,3 +270,72 @@ def structural_similarity(im1, im2,
             return mssim, S
         else:
             return mssim
+
+
+def multiscale_structural_similarity(im1, im2,
+                          *,
+                          win_size=11,
+                          multiscale_weights=(0.0448, 0.2856, 0.3001, 0.2363, 0.1333),
+                          # \alpha, \beta values from https://ece.uwaterloo.ca/~z70wang/publications/msssim.pdf
+                          channel_axis = 2,
+                          **kwargs):
+    """
+    Compute the multiscale structural similarity index between two images.
+
+    Parameters
+    ----------
+    im1, im2 : ndarray
+        Images. Any dimensionality with same shape.
+    win_size : int, optional
+        The side-length of the sliding window used in comparison. Must be an
+        odd value.
+    channel_axis : int
+        This parameter indicates which axis of the array corresponds
+        to channels.
+    multiscale_weights : list, optional
+        The weights that needs to applied to the ssim at each scale
+
+    Returns
+    -------
+    ms_ssim : float
+        The multiscale structural similarity index over the image.
+    
+    Notes
+    -----
+    - kwargs can be used to pass arguments for the ssim metric at each scale see metrics.ssim for details
+    - Code adapted version of the pytorch implementation of the msssim from `pytorch-mssim` by @jorge-pessoa[1]
+    Reference:
+        [1] https://github.com/jorge-pessoa/pytorch-msssim/blob/dev/pytorch_msssim/__init__.py
+            (MIT License)       
+
+    """
+    check_shape_equality(im1, im2)
+
+    mssim = []
+    mcs = []
+    
+    for weight in multiscale_weights:
+        # calculate ssim at current scale
+        if min(im1.shape[0], im1.shape[1]) > 1:
```

Comment:
shouldn't `win_size` be the smallest allowed image size?

Reply:
Oops! absolutely you are right!